compiler/AST/symbol.cpp

```diff
@@ -379,6 +379,84 @@ SymExpr* Symbol::getSingleDef() const {
 }
 
 
+Expr* Symbol::getInitialization() const {
+  // In theory, this should be the first "def" for the symbol,
+  // but that might be obfuscated by PRIM_ADDR_OF.
+
+  FnSymbol* fn = toFnSymbol(defPoint->parentSymbol);
+  ModuleSymbol* mod = toModuleSymbol(defPoint->parentSymbol);
+  if (fn == NULL && mod != NULL ) {
+    // Global variables are initialized in their module init function
+    fn = mod->initFn;
+  }
+
+  Expr* stmt;
+  // We'll search statements starting with stmt for the one
+  // initializing our variable.
+  if (mod != NULL) {
+    stmt = fn->body->body.head;
+  } else {
+    stmt = defPoint->getStmtExpr()->next;
+  }
+
+  const Symbol *curSym = this;
+  const Symbol *refSym = NULL;
+
+  while (stmt != NULL) {
+    std::vector<SymExpr*> symExprs;
+    collectSymExprs(stmt, symExprs);
+
+    bool isDef = false;
+    bool isUse = false;
+
+    for_vector(SymExpr, se, symExprs) {
+      Symbol* sym = se->symbol();
+      if (sym == curSym || sym == refSym) {
+        int result = isDefAndOrUse(se);
+        isDef |= (result & 1);
+        isUse |= (result & 2);
+      }
+    }
+
+    if (isDef || isUse) {
+
+      if (isDef) {
```

Comment:
Could you hoist this out of the parent if-statement, and instead structure the code like this?
```cpp
if (isDef) {
  // return ...
} else if (isUse) {
  // ...
}
```